From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Mon, 8 Jan 2024 19:26:23 +0100
Subject: [PATCH] Modifiable Bukkit enums - Complete legacy conversion -
 Conversion is explicitly for item or block - Expose Bukkit API

Has `CraftMagicNumbers` implement `UnsafeValues.fromLegacy(Material, boolean`),
which is defined in the "Modifiable Bukkit enums - Complete legacy conversion - Conversion is explicitly for item or block - Expose Bukkit API" patch in the `fiddle-api` module.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
index 2b385909268285140b58c01b03f57e4422486612..6a8e622aa58fea296a293ab9b654f6194d28b71f 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
@@ -235,6 +235,13 @@ public final class CraftMagicNumbers implements UnsafeValues {
         return this.fromLegacy(material, false); // Fiddle - modifiable Bukkit enums - Material - make legacy conversion complete - conversion is explicitly for item or block
     }
 
+    // Fiddle end - modifiable Bukkit enums - Material - make legacy conversion complete
+    @Override
+    public Material fromLegacy(Material material, boolean itemPriority) {
+        return CraftLegacy.fromLegacy(material, itemPriority);
+    }
+    // Fiddle end - modifiable Bukkit enums - Material - make legacy conversion complete
+
     @Override
     public Material fromLegacy(MaterialData material) {
         return this.fromLegacy(material, false); // Fiddle - modifiable Bukkit enums - Material - make legacy conversion complete - conversion is explicitly for item or block
